### PR TITLE
Add projects page with a small portfolio of work.

### PR DESCRIPTION
**Pull request by jamohile**
** Demo **
https://jamohile-step-2020.uc.r.appspot.com/projects.html

**Done**
- Added a page (Projects) with a list of projects, and related descriptions/tags.
- Allow filtering by tag(s)

**Notable Design Decisions**
- To maintain 0 dependency on external libraries, using html template tag along with ES6 classes to allow client-side components.
- Very loosely modeled component-api off of React.

** To Do**
- Explore SASS styling, or some other way to break up the monolithic styles.css
- Refactor list of projects (or put it online with datastore)
- Dynamically generate tags based on projects
- Refactor Tag/Component classes using a superclass, this is essentially making a reusable mini-framework for the site.

### CODE REVIEW

File: portfolio/src/main/webapp/style.css [hunk 1]
```diff
@@ -123,37 +159,113 @@ main#landing > #hero{
...
-    }
+main#projects > #projects-list > .project > .tags {
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+
+    margin-top: 16px;
+}
+
+main#projects > #projects-list > .project > .tags > .tag {
```
Comment (mctavish) [inline portfolio/src/main/webapp/style.css:213]: This is fine, but I do appreciate your desire for something that reads a bit cleaner.

File: portfolio/src/main/webapp/js/projects.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        }
+    }
+}
+
+/**
+ * A project is a single portfolio item.
+ * The Project class handles displaying projects while applying necessary filters.
+ */
+
+class Project {
```
Comment (mctavish) [inline portfolio/src/main/webapp/js/projects.js:88]: Nicely done

File: portfolio/src/main/webapp/style.css (hunk 1 above)
Comment (jamohile) [reply portfolio/src/main/webapp/style.css:213]: I imagine those selectors will get old pretty fast :) Will migrate to SASS!

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -10,13 +10,13 @@
...
   <body>
     <header>
-        <div id="wordmark">Jay Mohile</div>
+        <div id="wordmark" href="/">Jay Mohile</div>
         <nav>
-            <a href="/personal">
+            <a href="/personal.html">
                 <div class="hover-indicator"></div>
                 <div class="title">Personal</div>
             </a>
```
[the remark this answers is not in the document]
Comment (jamohile) [reply portfolio/src/main/webapp/index.html:18]: Got it, will do from now on.

**Pull request merged**